python/plugins/processing/ProcessingPlugin.py

```diff
@@ -282,6 +292,111 @@ def initGui(self):
         self.iface.actionToggleEditing().triggered.connect(partial(self.sync_in_place_button_state, None))
         self.sync_in_place_button_state()
 
+        # Sync project models
+        self.projectModelsMenu = None
+        self.projectMenuAction = None
+        self.projectMenuSeparator = None
+
+        self.projectProvider = QgsApplication.instance().processingRegistry().providerById("project")
+        self.projectProvider.algorithmsLoaded.connect(self.updateProjectModelMenu)
+
+    def updateProjectModelMenu(self):
+        """Add projects models to menu"""
+
+        if self.projectMenuAction is None:
+            self.projectModelsMenu = QMenu(self.tr("Models"))
+            self.projectMenuAction = self.iface.projectMenu().insertMenu(self.iface.projectMenu().children()[-1], self.projectModelsMenu)
+            self.projectMenuAction.setParent(self.projectModelsMenu)
+            self.iface.projectMenu().insertSeparator(self.projectMenuAction)
+
+        self.projectModelsMenu.clear()
+
+        for model in self.projectProvider.algorithms():
+            modelSubMenu = self.projectModelsMenu.addMenu(model.name())
+            modelSubMenu.setParent(self.projectModelsMenu)
+            action = QAction(self.tr("Execute…"), modelSubMenu)
+            action.triggered.connect(partial(self.executeProjectModel, model.id(), self.projectModelsMenu, self.toolbox.in_place_mode))
+            modelSubMenu.addAction(action)
+            if model.flags() & QgsProcessingAlgorithm.FlagSupportsBatch:
+                action = QAction(self.tr("Execute as Batch Process…"), modelSubMenu)
+                modelSubMenu.addAction(action)
+                action.triggered.connect(partial(self.executeProjectModel, model.id(), self.projectModelsMenu, self.toolbox.in_place_mode, True))
+
+    @pyqtSlot(str, QWidget, bool, bool)
+    def executeProjectModel(self, alg_id, parent, in_place=False, as_batch=False):
```

Comment:
Last request - can you fix the name of this function? It's not always for project models now

Reply:
Done!